cmd/fleet/serve.go

```diff
@@ -385,43 +381,79 @@ the way that the Fleet server works.
 				}
 			}
 
-			var (
-				scepStorage      *scep_mysql.MySQLDepot
-				depStorage       *mysql.NanoDEPStorage
-				mdmStorage       *mysql.NanoMDMStorage
-				mdmPushService   *nanomdm_pushsvc.PushService
-				mdmPushCertTopic string
-			)
-			if config.MDMApple.Enable {
-				if err := config_apple.Verify(config.MDMApple); err != nil {
-					initFatal(err, "verify apple mdm config")
+			if config.MDM.AppleAPNsCert != "" || config.MDM.AppleAPNsKey != "" || config.MDM.AppleSCEPCert != "" || config.MDM.AppleSCEPKey != "" {
+				// if any of the 4 flags are set and not all of them, this is an error
+				checks := map[string]string{
+					"mdm_apple_apns_cert": config.MDM.AppleAPNsCert,
+					"mdm_apple_apns_key":  config.MDM.AppleAPNsKey,
+					"mdm_apple_scep_cert": config.MDM.AppleSCEPCert,
+					"mdm_apple_scep_key":  config.MDM.AppleSCEPKey,
 				}
-				scepStorage, err = mds.NewMDMAppleSCEPDepot(
-					[]byte(config.MDMApple.SCEP.CA.PEMCert),
-					[]byte(config.MDMApple.SCEP.CA.PEMKey),
-				)
-				if err != nil {
-					initFatal(err, "initialize mdm apple scep storage")
+				for flag, value := range checks {
+					if value == "" {
+						initFatal(fmt.Errorf("missing flag --%s", flag), "validate Apple MDM")
+					}
 				}
-				mdmStorage, err = mds.NewMDMAppleMDMStorage(
-					[]byte(config.MDMApple.MDM.PushCert.PEMCert),
-					[]byte(config.MDMApple.MDM.PushCert.PEMKey),
-				)
+
+				apnsCert, err := config.MDM.AppleAPNs()
 				if err != nil {
-					initFatal(err, "initialize mdm apple MySQL storage")
+					initFatal(err, "validate Apple APNs certificate and key")
 				}
-				mdmPushCertTopic, err = cryptoutil.TopicFromPEMCert([]byte(config.MDMApple.MDM.PushCert.PEMCert))
-				if err != nil {
-					initFatal(err, "extract topic from mdm push certificate")
+				if _, err := config.MDM.AppleSCEP(); err != nil {
+					initFatal(err, "validate Apple SCEP certificate and key")
 				}
-				depStorage, err = mds.NewMDMAppleDEPStorage([]byte(config.MDMApple.DEP.Token))
-				if err != nil {
-					initFatal(err, "initialize mdm apple dep storage")
+
+				const (
+					apnsConnectionTimeout = 10 * time.Second
+					apnsConnectionURL     = "https://api.sandbox.push.apple.com"
+				)
+
+				// check that the Apple APNs certificate is valid to connect to the API
+				ctx, cancel := context.WithTimeout(context.Background(), apnsConnectionTimeout)
```

Comment:
Not sure I understand your point here - do you mean that you feel the timeout is too short? I used the same one we use to validate the TLS in `fleetctl debug connection`.

Reply:
Right, but that is a command that is ran by a person. This is ran by a service that spawns up fleet instances. There's usually a startup time that is set, let's assume this is fine unless proven otherwise.

Reply:
I still don't understand what we're talking about here - i.e. what was your initial concern :laughing: Timeout too long/too short? This is just run at startup, not at regular intervals. We have a number of network connections being established at fleet startup, I'm not sure what the concern is for that specific one?

Reply:
Too long startup time and have the orchestrator kill the service cause healthz doesn't respond, entering a loop.

Reply:
The redis tls handshake timeout is also 10s by default, so by itself this check doesn't introduce anything worse that could already happen - the accumulation of all those checks could become an issue, but as you mention, let's assume it's fine for now, we'll see in dogfood (or load testing) if this actually causes problems.